packages/typed-urls/src/lib/resources/utils/interpolate/interpolate.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @param str
+ * @param params
+ * @returns Interpolation of the path with the parameters.
+ */
+export function interpolate(str: string, params = {}): string {
+  const paramsCopy = { ...params };
+  const result: string[] = [];
+  let segmentMatch;
+  let key: string | null;
+
+  (str || "").split(":").forEach((segment: string, i: number) => {
+    if (i === 0 || !/^[a-z]/i.test(segment)) {
+      result.push(i === 0 ? segment : ":" + segment);
+    } else {
+      segmentMatch = segment.match(/(\w+)(?:[?*])?(.*)/);
+      // @ts-ignore
+      key = segmentMatch[1];
+      // @ts-ignore
+      result.push(paramsCopy[key]);
+      // @ts-ignore
+      result.push(segmentMatch[2] || "");
+      // @ts-ignore
+      delete paramsCopy[key];
```

Comment:
Is it required? delete operation is very slow

Reply:
Truly... I won't change the code of the library because I didn't write it. Let's focus on the structure and build process